### PR TITLE
Added roles configuration screens

### PR DESCRIPTION
**Pull request by reinaris**
I added screen to configure roles trough the admin.
Basic controller and templates.

Protected the user from deleting or editing the default admin and user role by adding some code to ability.rb.

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://reinaris-roles-configuration.spree-spree-9ed724b77adec04c49c2.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://reinaris-roles-configuration.spree-spree-9ed724b77adec04c49c2.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

File: core/app/models/spree/ability.rb [hunk 1]
```diff
@@ -64,6 +63,9 @@ def initialize(user)
         ability = clazz.send(:new, user)
         @rules = rules + ability.send(:rules)
       end
+
+      # Protect admin and user roles
+      cannot [:update, :destroy], Role, name: ['admin', 'user']
```
Comment (JDutil) [inline core/app/models/spree/ability.rb]: I think admin is all we need to protect.  The user role isn't actually used anywhere as far as I know.

**Comment by JDutil** [comment]
:+1: Could use a feature spec, but looks good.

**Comment by reinaris** [comment]
Will add some feature specs!

**Pull request closed without merging**